src/bridges/nft-basic/NftVault.sol

```diff
@@ -0,0 +1,130 @@
+     * @dev This method can only be called from the RollupProcessor.sol. The first step of the
+     * deposit flow returns a virutal asset note that will represent the NFT on Aztec. After the
+     * virutal asset note is received on Aztec, the user calls matchDeposit which deposits the NFT
+     * into Aztec and matches it with the virtual asset. When the virutal asset is sent to this function
+     * it is burned and the NFT is sent to the user (withdraw).
+     *
+     * @param _inputAssetA - ETH (Deposit) or VIRTUAL (Withdrawal)
+     * @param _outputAssetA - VIRTUAL (Deposit) or 0 ETH (Withdrawal)
+     * @param _totalInputValue - must be 1 wei (Deposit) or 1 VIRTUAL (Withdrawal)
+     * @param _auxData - corresponds to the Ethereum address id in the AddressRegistry.sol for withdrawals
+     * @return outputValueA - 1 VIRTUAL asset (Deposit) or 0 ETH (Withdrawal)
+     *
+     */
+
+    function convert(
+        AztecTypes.AztecAsset calldata _inputAssetA,
+        AztecTypes.AztecAsset calldata,
+        AztecTypes.AztecAsset calldata _outputAssetA,
+        AztecTypes.AztecAsset calldata,
+        uint256 _totalInputValue,
+        uint256,
+        uint64 _auxData,
+        address
+    )
+        external
+        payable
+        override (
+            // uint64 _auxData,
+            // address _rollupBeneficiary
+            BridgeBase
+        )
+        onlyRollup
+        returns (uint256 outputValueA, uint256 outputValueB, bool isAsync)
+    {
+        if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.NOT_USED
+                || _inputAssetA.assetType == AztecTypes.AztecAssetType.ERC20
+        ) revert ErrorLib.InvalidInputA();
+        if (
+            _outputAssetA.assetType == AztecTypes.AztecAssetType.NOT_USED
+                || _outputAssetA.assetType == AztecTypes.AztecAssetType.ERC20
+        ) revert ErrorLib.InvalidOutputA();
+        if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.ETH
+                && _outputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL
+        ) {
+            if (_totalInputValue != 1 wei) {
+                revert ErrorLib.InvalidInputAmount();
+            }
+            return (1, 0, false);
+        } else if (
+            _inputAssetA.assetType == AztecTypes.AztecAssetType.VIRTUAL
+                && _outputAssetA.assetType == AztecTypes.AztecAssetType.ETH
+        ) {
+            NftAsset memory token = tokens[_inputAssetA.id];
+            if (token.collection == address(0x0)) {
+                revert ErrorLib.InvalidInputA();
+            }
+
+            address _to = REGISTRY.addresses(_auxData);
+            if (_to == address(0x0)) {
+                revert ErrorLib.InvalidAuxData();
+            }
+
+            IERC721(token.collection).transferFrom(address(this), _to, token.id);
+            emit NftWithdraw(_inputAssetA.id, token.collection, token.id);
+            delete tokens[_inputAssetA.id];
```

Comment:
Not following-checks-effects interaction. For calls made from inside `transferFrom` the `nftAsset` would still be there.